pepper-apis/src/main/java/org/broadinstitute/ddp/service/WorkflowService.java

```diff
@@ -75,4 +90,29 @@ public WorkflowResponse buildStateResponse(Handle handle, String userGuid, Workf
         }
         return WorkflowResponse.unknown();
     }
+
+    private void createActivityInstanceIfMissing(
+            Handle handle,
+            WorkflowState fromState,
+            WorkflowState nextState,
+            String operatorGuid,
+            String userGuid
+    ) {
+        if (nextState.getType() != StateType.ACTIVITY) {
+            return;
+        }
+        ActivityState activityState = (ActivityState) nextState;
+        String instanceGuid = handle.attach(JdbiActivityInstance.class)
+                .findLatestInstanceGuidByUserGuidAndActivityId(userGuid, activityState.getActivityId())
+                .orElse(null);
+        boolean activityInstanceIsMissing = instanceGuid == null;
+        if (activityInstanceIsMissing) {
```

Comment:
Should we also check that the activity definition (in `study_activity` table) has `max_instances_per_user` is `== null || > 0`? If it's zero it means we can't have any instances. I doubt we'll see that in practice, but just in case?

Reply:
Thanks, that's a really important point!

Reply:
Done.